src/Provider/Exception/IdentityProviderException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace League\OAuth2\Client\Provider\Exception;
+
+class IdentityProviderException extends \Exception
+{
+    protected $result;
+
+    public function __construct($message, $code, $result)
+    {
+        $this->result = $result;
```

Comment:
$result is same with argument passed to errorCheck. So it is an array. Sometimes the error response have information other than error message and code. So we need make the user can access it on the exception.